src/frontends/tensorflow/src/op/while.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "input_model.hpp"
+#include "op_table.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_while_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"While", "StatelessWhile"});
+    auto node_name = node.get_name();
+    auto translate_session = node.get_translate_session();
+    auto input_size_t = node.get_input_size();
+    auto input_size = static_cast<int>(input_size_t);
+
+    TENSORFLOW_OP_VALIDATION(node,
+                             translate_session,
+                             "[TensorFlow Frontend] Internal error: Translate session is nullptr.");
+    // retrieve condition and body graphs
+    auto cond_type = node.get_attribute<std::string>("cond");
+    auto body_type = node.get_attribute<std::string>("body");
+    auto cond_model = translate_session->get_body_ov_model(cond_type);
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        cond_model,
+        "[TensorFlow Frontend] Internal error or incorrect input model. Cannot find body graph with name " + cond_type);
+    auto body_model = translate_session->get_body_ov_model(body_type);
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        body_model,
+        "[TensorFlow Frontend] Internal error or incorrect input model. Cannot find body graph with name " + body_type);
+
+    // inject condition body graph prior to Loop node
+    // to check condition before to start iterations
+    ov::OutputVector ov_inputs;
+    for (int input_ind = 0; input_ind < input_size; ++input_ind) {
+        ov_inputs.push_back(node.get_input(input_ind));
+    }
+    auto cond_prior = cond_model->clone();
```

Comment:
It looks like every time you call `inject_body_model` you will do copy of the body, can it be done inside the `inject_body_model` function?

Reply:
Let us think this sort of improvement in different PR because it will require changes in other files.
This change only for convenience to have code by one line shorter. Not critical